src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder.cs

```diff
@@ -167,9 +192,12 @@ private ImmutableArray<BoundPatternSwitchSection> BindPatternSwitchSections(Boun
                             hasErrors = true;
                         }
 
+                        // We always treat the default label as reachable, even if the switch is complete.
```

Comment:
Because it is required to be compatible with existing code. When you switch on a bool and have cases for true, false, and default, all are reachable. More generally, it is necessary to enable people to write switches that are resilient to extensions of the underlying data types. Your switch might be complete today, but you want to add a default case to handle "anything else" that might be added in the future. This lets you do that without a warning.